File: student.py
class Student:
    def __init__(self, name, Id, percentage = 0, skills = []):
        self.name = name
        self.Id = Id
        self.percentage = percentage
        self.skills = skills
    
    #Getter methods
    
    def get_name(self):
        return self.name

 

    def get_Id(self):
        return self.Id

 

    def get_percentage(self):
        return self.percentage

 

    def get_skills(self):
        return self.skills

 

    #Setter methods
    
    def set_name(self, name):
        self.name = name

 

    def set_percentage(self, perct):
        self.percentage = perct

 

    def set_skill(self, skillsnew):
        self.skills = skillsnew

 

        
class Operation (Student):
    def __init__(self):
        Student.__init__(self,"saurav",1,10,["java"])
    
    #Method to change name of a given object
    def changeName(self, newName):
        self.set_name(newName)

 

    #Method to change name according to user input
    def changeNameUserInput(self):
        newName = input("Enter your new Name")
        self.set_name(newName)

 

     #Method to add upto 5 new skills   
    def addSkills(self,newSkills):
        if newSkills.length < 5:
            newarr = newSkills+ self.skills
            self.set_skill(newarr)
     

File: Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PairProg
{
    public class Student
    {
        string name;
        int id;
        double percentage;
        string []skills;
        int skillsSize;

        public Student()
        {
            name = "";
            id = 0;
            percentage = 0;
            skills = null;
            skillsSize = 0;
        }

        public Student(String name = "", int id = 0, double percentage = 0, string[] skills = null, int skillsSize = 0)
        {
            this.name = name;
            this.id = id;
            this.percentage = percentage;
            this.skills = skills;
            this.skillsSize = skillsSize;
        }

        public void setName(String name)
        {
            this.name = name;
        }
        public void setId(int id)
        {
            this.id = id;
        }
        public void setPercentage(double percentage)
        {
            this.percentage = percentage;
        }
        public void setSkills(string []skills)
        {
            this.skills = skills;
        }
        public void setSkillsSize(int skillsSize)
        {
            this.skillsSize = skillsSize;
        }

        public string getName() { return name; }
        public int getId() { return id; }
        public double getPercentage() { return percentage; }
        public string[] getSkills() { return skills; }
        public int getSkillsSize() { return skillsSize; }


    }
}
